File: frontend/src/api/ai.ts
import request from '@/utils/request'

export interface ClassifyResult {
  category: string
  priority: string
  reason: string
  suggested_tags: string[]
}

// AI 智能分类
export const classifyTask = (title: string, description?: string) => {
  return request.post('/ai/classify', { title, description }) as Promise<ClassifyResult>
}

File: backend/app/schemas/ai.py
from pydantic import BaseModel, Field
from typing import Optional


class TaskClassifyRequest(BaseModel):
    """AI 分类请求"""
    title: str = Field(..., min_length=1, max_length=200)
    description: Optional[str] = None


class TaskClassifyResponse(BaseModel):
    """AI 分类响应"""
    category: str
    priority: str
    reason: str
    suggested_tags: list[str] = Field(default_factory=list)
